### PR TITLE
RHICOMPL-187 - GraphQL refactor into classes

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

**Comment by dLobatog** [comment]
@akofink - work in progress

**Comment by akofink** [comment]
lol, I mean, why is this WIP?

**Comment by dLobatog** [comment]
@akofink Oh :laughing: - there are a few syntax errors, cops to be fixed, and I want to write some tests.

File: app/graphql/types/query.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    field :system, Types::System, null: true do
+      description 'Details for a system'
+      argument :id, String, required: true
+    end
+
+    def system(args)
+      Pundit.authorize(context[:current_user], ::Host.find(args[:id]), :show?)
+    end
+
+    field :allImageStreams, [Types::System], null: true do
```
Comment (akofink) [inline app/graphql/types/query.rb]: Should this be snake_case?

File: Gemfile.lock [hunk 2]
```diff
@@ -1,389 +0,0 @@
-GEM
```
[the remark this answers is not in the document]
Comment (dLobatog) [reply Gemfile.lock]: Mistake, sorry

File: test/graphql/queries/system_query_test.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    hosts(:one).profiles << profiles(:one)
+    hosts(:one).profiles << profiles(:two)
+
+
+    result = Schema.execute(
+      query,
+      variables: { policyId: profiles(:one).id },
+      context: { current_user: users(:test) }
+    )
+    binding.pry
```
Comment (akofink) [inline test/graphql/queries/system_query_test.rb]: :beetle:

Comment (akofink) [reply test/graphql/queries/system_query_test.rb]: How did the tests pass, I wonder?

File: app/graphql/types/query.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    field :profile, Types::Profile, null: true do
+      argument :id, String, required: true
+    end
+
+    field :business_objectives, [Types::BusinessObjective], null: true do
+      description 'All business objectives visible by the user'
+    end
+
+    def all_systems(args)
```
[the remark this answers is not in the document]
Comment (dLobatog) [reply app/graphql/types/query.rb]: Updated, good catch

File: app/graphql/types/system.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    def rules_passed(profile_id:)
+      object.rules_passed(::Profile.find_by(id: profile_id))
+    end
+
+    def rules_failed(profile_id:)
+      object.rules_failed(::Profile.find_by(id: profile_id))
+    end
+
+    def rule_objects_failed(host)
```
Comment (akofink) [inline app/graphql/types/system.rb]: Where is the arg for this defined? I don't see it around 26 above.

**Comment by akofink** [comment]
I figured it out - this is caused by some missing camelCase attributes in the [frontend](https://github.com/RedHatInsights/compliance-frontend/pull/157#pullrequestreview-256475818).

**Pull request merged**